eng/liveBuilds.targets

```diff
@@ -78,62 +78,68 @@
-    </PropertyGroup>
-    <ItemGroup>
-      <CoreCLRFiles Include="$(CoreCLRSharedFrameworkDir)*.*" />
-      <CoreCLRCrossTargetFiles Condition="'$(CoreCLRCrossTargetComponentDir)' != ''" Include="$(CoreCLRCrossTargetComponentDir)*.*" IsNative="true" />
-      <CoreCLRFiles Include="$(CoreCLRArtifactsPath)Redist/**/*.dll" />
-      <CoreCLRFiles>
-        <IsNative>true</IsNative>
-      </CoreCLRFiles>
-      <_systemPrivateCoreLib Include="$(CoreCLRArtifactsPath)System.Private.CoreLib.dll"
-                             Condition="Exists('$(CoreCLRArtifactsPath)System.Private.CoreLib.dll')" />
-      <_systemPrivateCoreLib Include="$(CoreCLRArtifactsPath)IL/System.Private.CoreLib.dll"
-                             Condition="Exists('$(CoreCLRArtifactsPath)IL/System.Private.CoreLib.dll') and '@(_systemPrivateCoreLib)' == ''" />
-      <CoreCLRFiles Include="@(_systemPrivateCoreLib)" />
-      <CoreCLRFiles
-        Include="
-          $(CoreCLRSharedFrameworkDir)PDB/*.pdb;
-          $(CoreCLRSharedFrameworkDir)PDB/*.dbg;
-          $(CoreCLRSharedFrameworkDir)PDB/*.dwarf" />
-      <CoreCLRFiles
-        Include="$(CoreCLRArtifactsPath)PDB/System.Private.CoreLib.pdb;" />
-      <CoreCLRFiles Condition="Exists('$(CoreCLRArtifactsPath)PDB/System.Private.CoreLib.ni.pdb')"
-        Include="$(CoreCLRArtifactsPath)PDB/System.Private.CoreLib.ni.pdb" />
-      <CoreCLRCrossTargetFiles Condition="'$(CoreCLRCrossTargetComponentDir)' != ''"
-        Include="
-          $(CoreCLRCrossTargetComponentDir)PDB/*.pdb;
-          $(CoreCLRCrossTargetComponentDir)PDB/*.dbg;
-          $(CoreCLRCrossTargetComponentDir)PDB/*.dwarf" />
-
-      <CoreCLRCrossTargetFiles>
-        <TargetPath>runtime/$(CoreCLRCrossTargetComponentDirName)_$(TargetArchitecture)/native</TargetPath>
-      </CoreCLRCrossTargetFiles>
-    </ItemGroup>
-
-    <Error Condition="'@(CoreCLRFiles)' == ''" Text="The CoreCLR subset category must be built before building this project." />
-  </Target>
-
-  <Target Name="ResolveMonoFilesFromLocalBuild">
-    <Error Condition="!Exists('$(MonoArtifactsPath)')" Text="The Mono artifacts path does not exist '$(MonoArtifactsPath)'. The Mono subset category must be built before building this project." />
-
-    <ItemGroup>
-      <MonoFiles Include="$(MonoArtifactsPath)\*.*" />
-    </ItemGroup>
-
-    <Error Condition="'@(MonoFiles)' == ''" Text="The Mono subset category must be built before building this project." />
+  <Target Name="ResolveVMFilesFromLocalBuild">
+    <Error Condition="!Exists('$(CoreCLRArtifactsPath)') and '$(RuntimeFlavor)' == 'CoreCLR'" Text="The CoreCLR artifacts path does not exist '$(CoreCLRArtifactsPath)'. The CoreCLR subset category must be built before building this project." />
+    <Error Condition="!Exists('$(MonoArtifactsPath)') and '$(RuntimeFlavor)' == 'Mono'" Text="The Mono artifacts path does not exist '$(MonoArtifactsPath)'. The Mono subset category must be built before building this project." />
+
+    <Choose>
```

Comment:
I don't think this is legal in msbuild.  You have to have conditions on the `<PropertyGroup>` and `<ItemGroup>`.  MSBuild sometimes has weird rules.

Reply:
It seems to exist in other `.props` and `.targets` files as well as in `src/installer/Directory.Build.props`. Its definitely not a significant improvement over the organization of the code, but if I were to remove it, I would still prefer to have the property group+item group organized by conditions.

Reply:
It was brought to my attention that `<Choose>` currently only works at the project level. https://github.com/microsoft/msbuild/issues/2056